src/nvim/drawline.c

```diff
@@ -1126,7 +1223,7 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
       }
 
       if (draw_state == WL_FOLD - 1 && n_extra == 0) {
-        int fdc = compute_foldcolumn(wp, 0);
+        int fdc = statuscol.draw ? 0 : compute_foldcolumn(wp, 0);
```

Comment:
I would set `draw_state` directly to `WL_STC` and avoid mudding up the other blocks with `statuscol.draw`.

Either we go  

`WL_CMDLINE` -> `WL_STC` -> `WL_LINE` 

or

`WL_CMDLINE` -> `WL_FOLD` ->  `WL_SIGN` -> `WL_NR` -> `WL_LINE`

Reply:
Yeah that's cleaner, `WL_BRI` and `WL_SBR` are missing in those chains.